src/tag.cpp

```diff
@@ -282,16 +282,46 @@ int HSTag::cycleAllCommand(Input input, Output output)
     if (delta == 0) {
         return 0; // nothing to do
     }
-    FrameTree::CycleDelta cdelta = (delta == 1)
-            ? FrameTree::CycleDelta::Next
-            : FrameTree::CycleDelta::Previous;
-    bool succeeded = frame->cycleAll(cdelta, skip_invisible);
-    if (!succeeded) {
-        // we need to wrap. when cycling forward, we wrap to the beginning
-        FrameTree::CycleDelta rewind = (delta == 1)
-                    ? FrameTree::CycleDelta::Begin
-                    : FrameTree::CycleDelta::End;
-        frame->cycleAll(rewind, skip_invisible);
+    if (floating_focused()) {
+        int newIndex = static_cast<int>(floating_clients_focus_) + delta;
+        if (newIndex < 0) {
+            floating_focused = false;
+            frame->cycleAll(FrameTree::CycleDelta::End, skip_invisible);
+        } else if (static_cast<size_t>(newIndex) >= floating_clients_.size()) {
+            floating_focused = false;
+            frame->cycleAll(FrameTree::CycleDelta::Begin, skip_invisible);
+        } else {
+            floating_clients_focus_ = static_cast<size_t>(newIndex);
+            floating_clients_[floating_clients_focus_]->raise();
+        }
+    } else {
+        FrameTree::CycleDelta cdelta = (delta == 1)
+                ? FrameTree::CycleDelta::Next
+                : FrameTree::CycleDelta::Previous;
+        bool succeeded = frame->cycleAll(cdelta, skip_invisible);
+        if (!succeeded) {
```

Comment:
Add a comment here, what this means if cycleAll did not succeed? At least I had to think for a second what it means.
"End of tiling clients list, switch to float or rewind".
Or maybe a different name for the result would make it clearer.

Reply:
Is `focusChanged` clearer?

Reply:
It is indeed, thank you!